src/storage/mvcc/txn.rs

```diff
@@ -92,10 +120,16 @@ impl<S: Snapshot> MvccTxn<S> {
         self.writes.push(write);
     }
 
-    fn unlock_key(&mut self, key: Key) {
+    fn unlock_key(&mut self, key: Key, pessimistic: bool) -> ReleasedLock {
+        let released = if self.calc_lock_hash {
```

Comment:
I want to eliminate it when possible because the hash is calculated on raw key. It brings some overhead I think.

Reply:
Have you benchmarked? It does bring a lot of complexity and that means passing data around and more branching. It might go either way, but good to know.

Reply:
```
test storage::key::bench_index_key_gen_hash                            ... bench:          63 ns/iter (+/- 4)
test storage::key::bench_row_key_gen_hash                              ... bench:          36 ns/iter (+/- 3)
```
I will change the code  by always calculating it.😀